test/unit/dissemination-test.js

```diff
@@ -150,4 +151,89 @@ testRingpop('raise piggyback counter on issueAsSender', function t(deps, assert)
     assert.equal(disChangeSuspect.piggybackCount, 1, 'piggyback counter is raised');
     assert.equal(disChangeFaulty.piggybackCount, 1, 'piggyback counter is raised');
 
+});
+
+testRingpop('issueAsReceiver sets full sync correctly', function t(deps, assert) {
+    var membership = deps.membership;
+    var dissemination = deps.dissemination;
+
+    var addrAlive = '127.0.0.1:3001';
+    var incNo = Date.now();
+
+    // Clear changes to start fresh, otherwise local member changes
+    // recorded during bootstrap phase would have been issued.
+    dissemination.clearChanges();
+
+    var res = dissemination.issueAsReceiver(addrAlive, incNo,
+        membership.checksum);
+
+    assert.notOk(res.fullSync, 'full sync is false when checksums match');
+
+    res = dissemination.issueAsReceiver(addrAlive, incNo,
+        membership.checksum-1);
+    assert.ok(res.fullSync, 'full sync is true when checksums differ');
+});
+
+testRingpop('tryStartReverseFullSync keeps track of running jobs', function t(deps, assert) {
+    var dissemination = deps.dissemination;
+
+    dissemination._reverseFullSync = function(target, timeout, callback) {
+        assert.equal(dissemination.reverseFullSyncJobs, 1, 'reverse full sync jobs increased');
+        callback();
+
+        assert.equal(dissemination.reverseFullSyncJobs, 0, 'reverse full sync jobs decreased');
+    };
+
+
+    assert.plan(3);
+    assert.equal(dissemination.reverseFullSyncJobs, 0, 'running reverse full sync jobs is 0');
+
+    var target = '127.0.0.1:3001';
+    dissemination.tryStartReverseFullSync(target, 1000);
+});
+
+testRingpop('tryStartReverseFullSync doesn\'t start reverse full sync when out of workers', function t(deps, assert) {
+    var dissemination = deps.dissemination;
+    var ringpop = deps.ringpop;
+
+    ringpop.maxReverseFullSyncJobs = 0;
```

Comment:
I find this a bit unsatisfying. Can we see if this also works when this is not 0 and there is at least one job actually running?